### PR TITLE
Adding sender reports RTPC packets to indicate the stream sync.

### PR DESCRIPTION
**Pull request by QuinnDamerell-MS**

### CODE REVIEW

File: libftl/ftl_private.h [hunk 1]
```diff
@@ -77,7 +77,9 @@
 #define INGEST_PING_PORT 8079
 #define PEAK_BITRATE_KBPS 10000 /*if not supplied this is the peak from the perspective of the send buffer*/
 #define PING_TX_INTERVAL_MS 25
+#define SENDER_REPORT_TX_INTERVAL_MS 2000
```
Comment (sslivins) [inline libftl/ftl_private.h]: i think typically these are every 5 seconds

File: libftl/gettimeofday/gettimeofday.c [hunk 2]
```diff
@@ -145,3 +152,16 @@ float timeval_to_ms(struct timeval *tv) {
 
   return (float)(sec * 1000 + usec / 1000);
 }
+
+long long timeval_to_us(struct timeval *tv)
```
Comment (sslivins) [inline libftl/gettimeofday/gettimeofday.c]: this is declared as uint64_t in the header

File: libftl/media.c [hunk 3]
```diff
@@ -1213,40 +1212,98 @@ OS_THREAD_ROUTINE ping_thread(void *data) {
...
     }
 
-    gettimeofday(&ping->xmit_time, NULL);
-    _media_send_slot(ftl, &slot);
+    if (!ftl_get_state(ftl, FTL_DISABLE_TX_SENDER_REPORT))
+    {
+        uint64_t timeSinceLastSRSendMs = timeval_subtract_to_ms(&currentTime, &lastSenderReportSendTime_tv);
+        if (timeSinceLastSRSendMs > SENDER_REPORT_TX_INTERVAL_MS)
+        {
+            lastSenderReportSendTime_tv.tv_sec = currentTime.tv_sec;
```
Comment (sslivins) [inline libftl/media.c]: can just do lastSenderReportSendTime_tv = currentTime

**Review by sslivins — CHANGES_REQUESTED** [review]

File: libftl/ftl_private.h (hunk 1 above)
Comment (QuinnDamerell) [reply libftl/ftl_private.h]: I had that for testing and forgot.

**Comment by QuinnDamerell** [comment]
@sslivins please also double check my math on the gettimeofday functions as well. :)

**Review by sslivins — APPROVED** [review]
looks good

**Comment by QuinnDamerell-MS** [comment]
Thanks

**Pull request closed without merging**